### DISCUSSION TITLE
Full Magento 2 Support

### DISCUSSION THREAD

**Discussion by DavidLambauer** [beyondcode/herd-community#712] [General]
Although the docs say Laravel Herd would support Magento 2, the reality is that you have to run a few extra miles. I've been able to setup Magento 2, but ran into the following issues so far:

**Support for Opensearch, Varnish, and RabbitMQ** 

OpenSearch is mandatory to run Magento 2. There are even official Magento 2 versions that still support OpenSearch V1. RabbitMQ is similar. It's not a hard requirement, but you need it for a proper store. Magento 2 has built-in Varnish support, and it is highly recommended that Varnish be used as an FPC. 

Additionally, OpenSearch requires two Plugins to be installed. It'd be nice to do that through herd, too.

I had to install all three services locally, next to herd. 

**PHP Settings** 

A few mandatory PHP Ini Settings are required to be set differently. The `max_execution_time` can't be the default as Magento, in the developer mode, will render a ton of HTML on the fly. Having the `max_execution_time` next to the `memory_limit` setting would be nice.

**NGINX Settings**

Similar to the good old PHP Settings, we need to adjust a few `fastcgi_` params in nginx to solve common issues under Magento 2 Dev environments.

Under `~/Library/Application Support/Herd/config/nginx/herd.conf` add the following settings:

```ini
fastcgi_connect_timeout 1d;
fastcgi_read_timeout 1d;
fastcgi_send_timeout 1d;
```

Don't forget to `herd restart`

**Problems running tests due to Binary Location under macOS**

Herd stores all binaries under ` ~/Library/Application Support/Herd/bin`. Running tests via `bin/magento dev:tests:run integration` causes issues as the path is not escaped. An [issue was reported to magento](https://github.com/magento/magento2/issues/38707), but I doubt it'll be fixed soon. A potential workaround would be creating a hard symlink with a directory without spaces and aliasing PHP. 

Once I run into more issues/hickups, I will add them here.

**Comment by DavidLambauer** [comment]
Not directly related to Magento, but it's necessary to disable the two PHP Debug Settings for PHPStorm

![CleanShot 2024-06-05 at 10 33 58](https://github.com/beyondcode/herd-community/assets/1841317/39fabef9-8a4a-4e3c-962f-9f4b35d7a56a)

**Comment by justkidding96** [comment]
@mpociot Any update about this? Would be great if we have OpenSearch added to Herd so we can run Magento 2 from Herd instead of installing an extra service via homebrew.
